net/SslSocket.hpp

```diff
@@ -188,11 +207,13 @@ class SslStreamSocket final : public StreamSocket
                     return rc != 0;
             }
 
-            _doHandshake = false;
-
             if (rc == 1)
             {
                 // Successful handshake; TLS/SSL connection established.
+                _doHandshake = false;
+                _sslWantsTo = SslWantsTo::Neither; // No special needs until we are told otherwise.
```

Comment:
If we don't reset, we may do an extra `POLLOUT` when not needed, if we ended the handshake with WANTS_WRITE. This will get cleared out after the first `SSL_read` or `SSL_write`. But best to reset and have a clean start (though almost always we would have data to send, and we would therefore `POLLOUT`, first thing after the handshake completes, anyway).

Reply:
Looks sensible enough - proof is in the pudding; lets add that conditional & get it in =)